src/main.rs

```diff
@@ -462,6 +471,15 @@ fn parse_duration(arg: &str) -> anyhow::Result<Duration> {
         .map_err(|e| anyhow::anyhow!(e))
 }
 
+fn validate_nats_subject(subject: &str) -> anyhow::Result<()> {
+    let re = Regex::new(r"^(?:[A-Za-z0-9_-]+\.)*[A-Za-z0-9_-]+$").unwrap();
```

Comment:
_just_ in case, it would be good to have a small test that just ensures that this doesn't fail to instantiate, and have this at least be a `context(...)?` rather than an unwrap.

Reply:
I understand :+1: 
```rust
let re = Regex::new(r"^(?:[A-Za-z0-9_-]+\.)*[A-Za-z0-9_-]+$")
        .context("Failed to compile NATS subject regex")?;
```
So like this ?
and where exactly should I add a test ?

Reply:
That looks good! If you want, at the bottom of this file, you can add a:

```rust
#[cfg(test)]
mod test {
  fn test_validate_nats_subject() {
    ...
  }
}
```

Then that should run along with the cargo test suite

Reply:
I'll have to add this check for explicitly disallowing `*` and `>`
```rust
if re.is_match(subject) && !subject.contains('*') && !subject.contains('>')
```

Reply:
Gotcha :+1: I'll get on with it :rocket: